app/src/ingestion/pdf_postprocess.py

```diff
@@ -160,3 +160,35 @@ def group_texts(markdown_texts: list[EnrichedText]) -> list[EnrichedText]:
             grouped_texts.append(current_text)
 
     return grouped_texts
+
+
+def _should_merge_text(text: EnrichedText, next_text: EnrichedText) -> bool:
+    "Merges texts that are split across consecutive pages"
+    if text.headings != next_text.headings:
+        return False
+
+    assert text.page_number is not None
+    assert next_text.page_number is not None
```

Comment:
We could also log an error and then return False here -- defer to you on the desired behavior: bail out of the whole ETL pipeline (fail an assert) or skip this text but log that something unexpected happened

Reply:
At this step in the pipeline, all EnrichedText elements should have page numbers. If not, we should fail and fix the bug.